src/main/java/mekanism/common/content/sps/SPSMultiblockData.java

```diff
@@ -92,28 +119,25 @@ public void onCreated(Level world) {
         advancementArea = deathZone.inflate(15);
     }
 
-    private long getMaxInputGas() {
-        return MekanismConfig.general.spsInputPerAntimatter.get() * 2L;
-    }
-
     @Override
     public boolean tick(Level world) {
         boolean needsPacket = super.tick(world);
         double processed = 0;
         couldOperate = canOperate();
-        if (couldOperate && receivedEnergy > 0L) {
+        updateCachedRecipe();
+        if (cachedRecipe != null && couldOperate && receivedEnergy > 0L) {
             double lastProgress = progress;
-            final int inputPerAntimatter = MekanismConfig.general.spsInputPerAntimatter.get();
-            long inputNeeded = (inputPerAntimatter - inputProcessed) + inputPerAntimatter * (outputTank.getNeeded() - 1);
-            double processable = (double) receivedEnergy / MekanismConfig.general.spsEnergyPerInput.get();
+            lastReceivedRecipeAmount = cachedRecipe.getInput().amount();
```

Comment:
this variable name doesn't seem to match what it's used for - I only see it being set with the recipe's required amount (which should be updated in the cached recipe check)